ui/app/components/app/signature-request-original/signature-request-original.component.js

```diff
@@ -0,0 +1,325 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import ethUtil from 'ethereumjs-util'
+import classnames from 'classnames'
+import { ObjectInspector } from 'react-inspector'
+
+import { ENVIRONMENT_TYPE_NOTIFICATION } from '../../../../../app/scripts/lib/enums'
+import { getEnvironmentType } from '../../../../../app/scripts/lib/util'
+import Identicon from '../../ui/identicon'
+import AccountListItem from '../../../pages/send/account-list-item/account-list-item.component'
+import { conversionUtil } from '../../../helpers/utils/conversion-util'
+import Button from '../../ui/button'
+import { DEFAULT_ROUTE } from '../../../helpers/constants/routes'
+
+export default class SignatureRequestOriginal extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
+    metricsEvent: PropTypes.func,
+  }
+
+  static propTypes = {
+    balance: PropTypes.any,
+    cancel: PropTypes.func,
+    clearConfirmTransaction: PropTypes.func,
+    conversionRate: PropTypes.any,
+    history: PropTypes.object,
```

Comment:
Can we add `isRequired` as appropriate to these props OR add entries for `defaultProps`?

Reply:
Similarly for `contextTypes` above